### PR TITLE
Highlight line longer after jumping to file from Console

### PR DESCRIPTION
**Pull request by arthur801031**
Associated Issue: #1663 

### Summary of Changes

* Change the animation time from 1.5s to 4s.

### 4s Animation (animation-timing-function: normal):
rs4oktdqbx

### 1.5s Animation:
ff580476-d74a-11e6-880a-1181a022fcab
(GIF by @jnachtigall )

### CODE REVIEW

**Comment by arthur801031** [comment]
Hi @jasonLaster

Please check. Thank you! 🙂

**Comment by arthur801031** [comment]
I think with `animation-timing-function: ease-in` it's easier to see the highlighted line because a big chunk of 4s animation has a darker effect than the 4s animation with`animation-timing-function: normal`. Let me know which one you like more :)

### 4s Animation (animation-timing-function: ease-in):
4ox8kgfu1d

File: src/components/Editor/Editor.css [hunk 1]
```diff
@@ -98,7 +98,7 @@ html[dir="rtl"] .editor-mount {
 }
 
 .highlight-line .CodeMirror-line {
-  animation: fade-highlight-out 1.5s normal forwards;
+  animation: fade-highlight-out 4s normal forwards;
```
Comment (jasonLaster) [inline src/components/Editor/Editor.css]: hmm, i think we *only* want to do this when we're navigating from the console.

Perhaps:

```css
.highlight-line.ong .CodeMirror-line  { 
  animation: fade-highlight-out 3s normal forwards;
}
```

File: src/components/Editor/Editor.css [hunk 2]
```diff
@@ -98,7 +98,7 @@ html[dir="rtl"] .editor-mount {
 }
 
 .highlight-line .CodeMirror-line {
-  animation: fade-highlight-out 1.5s normal forwards;
+  animation: fade-highlight-out 3s normal forwards;
```
Comment (arthur801031) [inline src/components/Editor/Editor.css]: It seems a line would only be applied with the styles of `.highlight-line .CodeMirror-line` when navigating from the console. The navigation triggers the function at https://github.com/devtools-html/debugger.html/blob/master/src/components/Editor/index.js#L599.

The line is first cleared and highlighted under these 2 conditions: https://github.com/devtools-html/debugger.html/blob/master/src/components/Editor/index.js#L333-L349

Comment (jasonLaster) [inline src/components/Editor/Editor.css]: hmm - i think this also happens when you click on a breakpoint in the breakpoint list or you select a frame in the call stack...

Is that correct?

Comment (arthur801031) [inline src/components/Editor/Editor.css]: Hi @jasonLaster !

1. Clicking on a breakpoint in the breakpoint list would trigger `highlightLine()`, but the function returns right away, and it doesn't execute `this.editor.codeMirror.addLineClass(line - 1, "line", "highlight-line");`, which is highlighting the line.
yvxe4w1ual




2. Selecting a frame in the call stack would trigger `highlightLine()` and clear the line, but it doesn't call `this.editor.codeMirror.addLineClass(line - 1, "line", "highlight-line");`, so we won't see the line being applied with the styles of `.highlight-line .CodeMirror-line`.
2ped7795ob

**Comment by jasonLaster** [comment]
@arthur801031 this PR is probably good, but the internals of `highlightLine` are confusing and I'd like to get a better understanding of it before we move forward.

Perhaps we can add more integration tests here.

**Comment by nchevobbe** [comment]
@arthur801031 good job and good writeup ! 
I think we could do better with the animation though. Would you mind trying to : 
- keep it 4s long
- but start lowering down the opacity at 3s

Because right now the animation feels like it is in slow motion, and thus can trick the user to think the debugger "janks" and can't keep up.

I'd really appreciate if you could try that and put some GIF here :) Thanks ! 🙆

**Pull request closed without merging**